drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -4812,7 +4812,9 @@ Vector<uint8_t> RenderingDeviceVulkan::shader_compile_binary_from_spirv(const Ve
 		offset += sizeof(uint32_t);
 		memcpy(binptr + offset, &binary_data, sizeof(RenderingDeviceVulkanShaderBinaryData));
 		offset += sizeof(RenderingDeviceVulkanShaderBinaryData);
-		memcpy(binptr + offset, shader_name_utf.ptr(), binary_data.shader_name_len);
+		if (shader_name_utf.ptr()) {
```

Comment:
A size check (`binary_data.shader_name_len > 0`) is probably a more reasonable option (it is the only condition when pointer might be `nullptr`).

Reply:
Agreed, and then the `offset += binary_data.shader_name_len;` can likely be put under that same condition, as there's no point doing it if it's `+= 0`.

Reply:
It is irrelevant what the size is, you should never memcpy into nulltpr, that is why this fix was necessary in the first place. The memcpy already checks the the length.

Reply:
The string will never return a null pointer if the size is non-zero

Reply:
Yes it is pointless to call memcpy on a size equal 0, but that is not what this pull request about. Calling memcpy with zero copy-size is not a bug, calling memcpy with nullptr destination is a bug. It is not causing segmentation fault, just because the copy size is zero. There is a similar commit 9dd0d3f550a0940dd53833666cd082e81aab8eff about this. 
In addition making call on memcpy with zero size without additional "if" may be considered an optimization (because memcpy will check it anyways).

Reply:
The pointer won't be null though, so the test is equivalent

Reply:
>> why are you so against this?

I'm not. I'm just getting more confused on what my steps have to be here.

Reply:
I'm sorry you were down voting and arguing against it, that's why I assumed you were, what steps are you talking about?

Reply:
You still have three commits, you need to squash them together into one, and you have to update the commit message, see the instructions below, but I'll explain it here as well:
* You need to do a rebase with `git rebase -i` onto `master` (please update it to the latest upstream version as well as you're a bit behind), and put `fixup` in front of all your commits except the first one
* Then you need to do `git commit --amend` and update the message with a more descriptive one, see below
* Then you need to push the changes by `git push -f`

> there is no point to double check the size if memcpy does it.

Sure, but doing the size check *is equivalent* to doing the null check, as I've explained, and the memcpy *doesn't* do such a check, the behavior is undefined if either pointer is null, that's why the sanitizer fails, again, you can leave it as is and ignore the suggestion, or update it to use `binary_data.shader_name_len > 0` instead, it's your choice, but you're arguing from insufficient logic here so I'd suggest you follow it

Reply:
> EDIT: I can't get rid of the 3 commits with rebase though. I pick one, squash others and I still end up with more then one commit in the branch.

If you're fine with the above diff, I can do the rebase to squash the commits for you.

Reply:
Please do.

Reply:
Done. Added myself as co-author, not to take credit for the further tweaks, but because I also included the next `if` in the newly added one, so if I overdid the refactoring and something breaks, I'll take the blame.

Reply:
The commit is not linked to an account, as it took the first commit which was not

Reply:
Thanks, I fixed that up.